build.sbt

```diff
@@ -861,37 +864,72 @@ lazy val iceberg = (project in file("iceberg"))
   )
 // scalastyle:on println
 
-lazy val generateIcebergJarsTask = TaskKey[Unit]("generateIcebergJars", "Generate Iceberg JARs")
-
 lazy val icebergShaded = (project in file("icebergShaded"))
   .dependsOn(spark % "provided")
   .disablePlugins(JavaFormatterPlugin, ScalafmtPlugin)
   .settings (
     name := "iceberg-shaded",
     commonSettings,
     skipReleaseSettings,
-
-    // Compile, patch and generated Iceberg JARs
-    generateIcebergJarsTask := {
-      import sys.process._
-      val scriptPath = baseDirectory.value / "generate_iceberg_jars.py"
-      // Download iceberg code in `iceberg_src` dir and generate the JARs in `lib` dir
-      Seq("python3", scriptPath.getPath)!
-    },
-    Compile / unmanagedJars := (Compile / unmanagedJars).dependsOn(generateIcebergJarsTask).value,
-    cleanFiles += baseDirectory.value / "iceberg_src",
-    cleanFiles += baseDirectory.value / "lib",
+    // must exclude all dependencies from Iceberg that delta-spark includes
+    libraryDependencies ++= Seq(
+      // Fix Iceberg's legacy java.lang.NoClassDefFoundError: scala/jdk/CollectionConverters$ error
+      // due to legacy scala.
+      "org.scala-lang.modules" %% "scala-collection-compat" % "2.1.1" % "provided",
+      "org.apache.iceberg" % "iceberg-core" % "1.8.0" excludeAll (
+        icebergExclusionRules: _*
+      ),
+      "org.apache.iceberg" % "iceberg-hive-metastore" % "1.8.0" excludeAll (
+        icebergExclusionRules: _*
+      ),
+      "org.apache.hadoop" % "hadoop-client" % "2.7.3" excludeAll (
+        hadoopClientExclusionRules: _*
+      ),
+      "org.apache.hive" % "hive-metastore" % "2.3.8" excludeAll (
```

Comment:
isnt there a hive-metastore client just like hadoop-client? do you really the entirely of hive-metastore classes as an unshaded dependencies in delta-spark? 

cant we have minimize the dependencies further .. and move some of the stuff as test dependencies for running tests with real hive metastore which needs all the classes?

Reply:
see https://github.com/apache/iceberg/blob/main/build.gradle#L699

iceberg-hive-metastore uses hadoop client and hive-metastore as compile time dependency and thus the iceberg-hive-metastore jar does not have those classes. 

These added deps are not on delta-spark; they are on delta-iceberg. Either we go for the current way, or we make those as compile time dependency but user needs to include extra jars when they use it. I prefer the current way as uniform always need delta-iceberg jar, and uniform always needs those HMS classes to talk to HMS. 

I did not find "hadoop-client" equivalent... I choosed hive-metastore because iceberg-hive-metastore depends on it.